Creation modele donnee

SDdylan opened the pull request:
Created entity Trick, Group, User, Comment and Media and theirs Fixtures.

SDdylan committed:
Added entity: User, Trick, Comment, Group and Media

SDdylan committed:
Added password-hasher, fixture-bundle and fakerphp.
Added password to User class
Created Fixture for User

SDdylan committed:
Added fixtures for all classes,
Added password-hash to User fixture,
User class now implements UserInterface
Configured password-hash in services.yaml,
updated php version from 7.2.5 to 7.4.

jrusso-dev commented on migrations/Version20220223152655.php:
```diff
@@ -0,0 +1,37 @@
+<?php
```
Supprimer cette migration

jrusso-dev commented on migrations/Version20220223153917.php:
```diff
@@ -0,0 +1,36 @@
+<?php
```
Supprimer migration

jrusso-dev commented on src/DataFixtures/AppFixtures.php:
```diff
@@ -0,0 +1,17 @@
+<?php
```
Puisqu'aucune fixture n'existe dans ce fichier, il peut être supprimé

jrusso-dev commented on src/DataFixtures/UserFixtures.php:
```diff
@@ -0,0 +1,103 @@
+    public function load(ObjectManager $manager): void
+    {
+        $faker = Factory::create('fr_FR');
+        $randRole = ['["ROLE_ADMIN"]', '["ROLE_USER"]'];
```
Liste des rôles à déplacer dans l'entité User (après avoir créé User depuis la commande make:user)

jrusso-dev commented on src/DataFixtures/UserFixtures.php:
```diff
@@ -0,0 +1,103 @@
+    {
+        $faker = Factory::create('fr_FR');
+        $randRole = ['["ROLE_ADMIN"]', '["ROLE_USER"]'];
+        $typeMedia = ['image','video'];
```
Types de média à déplacer dans l'entité Media et mettre les valeurs dans des constantes

jrusso-dev commented on src/DataFixtures/UserFixtures.php:
```diff
@@ -0,0 +1,103 @@
+
+
+
+        for($i = 1; $i <= 4; $i++)
```
Ne pas utiliser de boucles pour créer 4 users, créer 1 user "standard" et 1 user admin

jrusso-dev commented on src/DataFixtures/UserFixtures.php:
```diff
@@ -0,0 +1,103 @@
+        for($i = 1; $i <= 4; $i++)
+        {
+            //Creation des utilisateurs
+            $user = new User();
```
Créer les 2 users (user/admin) depuis une méthode => buildUser(string $email, string $password, string $role)

jrusso-dev replied:
Forcer à la main l'email et le password pour pouvoir les retrouver facilement

jrusso-dev commented on src/DataFixtures/UserFixtures.php:
```diff
@@ -0,0 +1,103 @@
+            $manager->persist($user);
+
+            //creation des groupes
+            $group = new Group();
```
Créer 4 groupes à la main plutôt que dans une boucle

jrusso-dev commented on src/DataFixtures/UserFixtures.php:
```diff
@@ -0,0 +1,103 @@
+            $manager->persist($group);
+
+            //creation de trick
+            for($j=1; $j<=3; $j++)
```
Faire une boucle pour les figures (par exemple 20 figures)

jrusso-dev commented on src/DataFixtures/UserFixtures.php:
```diff
@@ -0,0 +1,103 @@
+                $trick->setTitle($faker->sentence(4))
+                      ->setDescription($faker->paragraph(4))
+                      ->setCreatedAt($faker->dateTime())
+                      ->setGrp($group)
```
Setter le groupe tiré de façon aléatoire

jrusso-dev commented on src/DataFixtures/UserFixtures.php:
```diff
@@ -0,0 +1,103 @@
+                      ->setDescription($faker->paragraph(4))
+                      ->setCreatedAt($faker->dateTime())
+                      ->setGrp($group)
+                      ->setUsers($user);
```
Setter le user tiré de façon aléatoire

jrusso-dev commented on src/DataFixtures/UserFixtures.php:
```diff
@@ -0,0 +1,103 @@
+            for($j=1; $j<=3; $j++)
+            {
+                $trick = new Trick();
+
```
`<?php

$group1 = 'Groupe 1';
$group2 = 'Groupe 2';
$group3 = 'Groupe 3';
$group4 = 'Groupe 4';

$randomGroup = rand(1, 4);

$randomUser = rand(1, 2);

$group = ${"group".$randomGroup};
$user = $randomUser === 1 ? $simpleUser : $adminUser;`

SDdylan committed:
Deleted User class and User Repository to use make:user function.
Deleted Migrations.

SDdylan committed:
Installed symfony/maker-bundle to use "make:user".
Added User entity and repository.
Added isVerified boolean in Comment.php.
Added CONST for image & video type in Media.php.
Refactored $users & $grp to $user & $group in Trick.php.
Modified Fixtures.

SDdylan committed:
Remove .idea from repo

SDdylan committed:
ignored .idea
